rotkehlchen/chain/ethereum/modules/yearn/decoder.py

```diff
@@ -93,97 +107,131 @@ def reload_data(self) -> None:
+            self._handle_withdraw_event(event=context.event, protocol=protocol)
+        else:  # a yearn contract method that we don't need to handle
             return FAILED_ENRICHMENT_OUTPUT
 
+        return TransferEnrichmentOutput(matched_counterparty=PROTOCOL_CPT_MAPPING[protocol])
+
+    def _handle_withdraw_event(self, event: 'EvmEvent', protocol: str) -> None:
+        """Decode an event associated with a withdrawal."""
+        counterparty = PROTOCOL_CPT_MAPPING[protocol]
         if (
-            is_deposit is True and
-            context.event.event_type == HistoryEventType.SPEND and
-            context.event.address == context.transaction.to_address
+            event.event_type == HistoryEventType.SPEND and
+            event.event_subtype == HistoryEventSubType.NONE and
+            event.address == ZERO_ADDRESS
         ):
-            context.event.event_type = HistoryEventType.DEPOSIT
-            context.event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
-            context.event.counterparty = protocol
-            vault_token_name = _get_vault_token_name(context.transaction.to_address)
-            context.event.notes = f'Deposit {context.event.balance.amount} {context.token.symbol} in {protocol} vault {vault_token_name}'  # noqa: E501
+            event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+            event.counterparty = counterparty
+            event.notes = f'Return {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} to a {counterparty} vault'  # noqa: E501
         elif (
-            is_deposit is True and
-            context.event.event_type == HistoryEventType.RECEIVE and
-            context.event.address == ZERO_ADDRESS
+            event.event_type == HistoryEventType.RECEIVE and
+            event.event_subtype == HistoryEventSubType.NONE and
+            event.address in self.vaults[protocol]
         ):
-            context.event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
-            context.event.counterparty = protocol
-            vault_token_name = _get_vault_token_name(context.transaction.to_address)
-            context.event.notes = f'Receive {context.event.balance.amount} {vault_token_name} after deposit in a {protocol} vault'  # noqa: E501
-        elif (
-            is_deposit is False and
-            context.event.event_type == HistoryEventType.RECEIVE and
-            context.event.address == context.transaction.to_address
+            event.event_type = HistoryEventType.WITHDRAWAL
+            event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+            event.counterparty = counterparty
+            vault_token_name = _get_vault_token_name(event.address)
+            event.notes = f'Withdraw {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} from {counterparty} vault {vault_token_name}'  # noqa: E501
+
+    def _handle_deposit_event(self, event: 'EvmEvent', protocol: str) -> None:
```

Comment:
you can limit thetype in the protocol too with `Literal`

```suggestion
    def _handle_deposit_event(self, event: 'EvmEvent', protocol: str) -> None:
```